From 65ea57969fcbd04938fa6dd669935f64246a7212 Mon Sep 17 00:00:00 2001
From: Daire McNamara <daire.mcnamara@microchip.com>
Date: Tue, 16 Feb 2021 11:29:45 +0000
Subject: [PATCH v1 1/2] dt-bindings: dma: microchip: Add Microchip PolarFire
 host binding

Add device tree bindings for uio driver for the Platform DMA (pdma)
on Microchip PolarFire SoC

Signed-off-by: Daire McNamara <daire.mcnamara@microchip.com>
---
 .../bindings/dma/microchip,mpfs-pdma-uio.yaml | 52 +++++++++++++++++++
 1 file changed, 52 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/dma/microchip,mpfs-pdma-uio.yaml

diff --git a/Documentation/devicetree/bindings/dma/microchip,mpfs-pdma-uio.yaml b/Documentation/devicetree/bindings/dma/microchip,mpfs-pdma-uio.yaml
new file mode 100644
index 000000000000..762b0123809f
--- /dev/null
+++ b/Documentation/devicetree/bindings/dma/microchip,mpfs-pdma-uio.yaml
@@ -0,0 +1,52 @@
+# SPDX-License-Identifier: GPL-2.0
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/dma/microchip,mpfs-pdma-uio.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Microchip PolarFire SoC UIO Platform DMA
+
+maintainers:
+  - Daire McNamara <daire.mcnamara@microchip.com>
+
+description: |
+  Platform DMA is a DMA engine of SiFive Unleashed. It supports 4
+  channels. Each channel has 2 interrupts. One is for DMA done and
+  the other is for DME error.
+
+  For designs using FPGA fabric PDMA is used to shuttle buffers
+  between user-space and FPGA fabric.
+
+  https://static.dev.sifive.com/FU540-C000-v1.0.pdf
+
+properties:
+  compatible:
+    items:
+      - const: microchip,mpfs-pdma-uio
+
+  reg:
+    maxItems: 1
+
+  interrupts:
+    minItems: 1
+    maxItems: 8
+
+  '#dma-cells':
+    const: 1
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - '#dma-cells'
+
+examples:
+  - |
+    dma@3000000 {
+      compatible = "microchip,mpfs-pdma-uio";
+      reg = <0x0 0x3000000 0x0 0x8000>;
+      interrupts = <23 24 25 26 27 28 29 30>;
+      #dma-cells = <1>;
+    };
+
+...
-- 
2.25.1

